### ISSUE TITLE
Where is traverse_latent_space_vae.py 

### ISSUE THREAD

**Issue by payphonePYF** [KingJamesSong/PDETraversal#2]
Hello! I'm very interested in your work. However, I've encountered an issue. I'm currently trying to evaluate the pre-trained VAE using mnist_eval.sh, but I've noticed that the repository doesn't contain the file "traverse_latent_space_vae.py," which is used in mnist_eval.sh.

**Comment by KingJamesSong** [comment]
Hi, thanks for your interest!

Sorry, I forgot to upload this file. I have uploaded the file "traverse_latent_space_vae.py" now, which is slightly modified from "traverse_latent_space.py" by changing the generator from GANs to VAEs.

There might be small mistakes about the names of variables (I changed some of them to improve the readability but I might not inspect the inconsistency that may happen somewhere). Please let me know if you have any issues.

BTW, if you focus on VAEs,  I would suggest you kindly check out our [another NeurIPS paper](https://github.com/KingJamesSong/latent-flow/tree/main), which proposes novel understandings of equivariance and disentanglement by exploring supervised and weakly-supervised settings.

**Comment by payphonePYF** [comment]
Hello, thank you for your response. Your other work is equally intriguing to me, and I will read it as soon as possible!

While comparing traverse_latent_space_vae.py and traverse_latent_space.py, I noticed the following:

In the line of code:
energy, shift = S.inference(dim, w if args_json.__dict__["shift_in_w_space"] else z, (step+1) * torch.ones(1, 1, requires_grad=True), G)
Both 'step+1' and 'step' are used in this step. Could you please clarify which version is correct?

Thank you!

**Comment by KingJamesSong** [comment]
Thanks for the correction! It should be 'step' in this work.

**Comment by payphonePYF** [comment]
Hi ,Your method is great, and I've conducted some experiments using your approach. 
I noticed that in  [traverse_latent_space.py](https://github.com/KingJamesSong/PDETraversal/blob/main/traverse_latent_space.py) there isn't a negative direction. I added the negative direction in a new [branch,](https://github.com/payphonePYF/PDETraversal/blob/main/traverse_latent_space.py) but I'm not sure if it's theoretically correct.

Additionally, I'm working on research related to chromosome  curvature disentanglement representation. However, I noticed in your paper that you mentioned, "since the curl of the gradient is known to be zero, potential flows are inherently irrotational and thus cannot model vorticity." Would it be possible for us to discuss this matter via email or another means?

**Comment by KingJamesSong** [comment]
Thanks! for your contribution and effort! Actually I wrote the two-direction traversal version in [traverse_latent_space_twosides.py](https://github.com/KingJamesSong/PDETraversal/blob/main/traverse_latent_space_twosides.py). Please check if it is consistent with your version.

I do not have chromosome background but this application seems very interesting to me! The explanation of curl of the gradient usually suits the non-symmetric convention of latent traversal works, but maybe it can be extended to symmetric traversal. Feel free to drop me an email to yue.song@unitn.it or songyue19960927@gmail.com

**Comment by payphonePYF** [comment]
Oh! Thank you for your reply. Yes, I missed that file, and it matches mine.
Thanks for providing your email address!Hope to have a nice email conversation with you.
